### ISSUE TITLE
Script Canvas Functions that contain a Container Variable crash the Editor when executed 

### ISSUE THREAD

**Issue by LB-TomaszSerwanski** [o3de/o3de#5926]
**Describe the bug**
Script Canvas Function that contains a Container Variable (i.e. Array, Map) crashes the Editor in runtime.

Please note that the Function does not need to have any node which uses such Variable, and the Editor still crashes.

Please also note, that sometimes, the crash occurs after entering the Game Mode for the second time, as the first one throws multiple errors and an assert into the Console (refer to the ErrorNoCrash.txt for more details).

Please refer to the attached video and log files for more details. 

**Steps to reproduce**
1. Open Script Canvas.
2. Create a new Function which contains a Container Variable with Scope set to Function (refer to the attached SC_Graphs.zip or Function.png).
3. Create a new Script which calls the above Function On Graph Start (refer to the attached SC_Graphs.zip or Script.png).
4. Assign the Script to an entity.
5. Enter Game Mode.
6. (optional) Exit Game Mode and enter it again.

**Expected behavior**
The Script executes correctly. The Editor remains stable.

**Actual behavior**
The Editor crashes.

**Assets required**

[SC_Graphs.zip](https://github.com/o3de/o3de/files/7602116/SC_Graphs.zip)

**Screenshots/Video**

![Function](https://user-images.githubusercontent.com/86953108/143424407-6dc4e307-8d09-4b59-b2e3-4b285157b1d5.png)

![Script](https://user-images.githubusercontent.com/86953108/143424397-63a4e1f0-f4b5-4bba-86c2-8172b31bcbc6.png)

https://user-images.githubusercontent.com/86953108/143424436-8b65b1e3-af90-4e29-a4d9-cbe8d7220c00.mp4

**Found in Branch**
Stabilization_2110 (99d0c39)

**Desktop/Device:**
 - Device: PC 
 - OS: Windows
 - Version: 10
 - CPU: AMD Ryzen 5 3600
 - GPU: Nvidia RTX 2060 SUPER
 - Memory: 16GB

**Call Stack Trace**
```
25) 00007FFF89E4B7F7 (KERNELBASE) : UnhandledExceptionFilter
24) 00007FFF8C5B51B0 (ntdll) : memset
23) 00007FFF8C59C766 (ntdll) : _C_specific_handler
22) 00007FFF8C5B20CF (ntdll) : _chkstk
21) 00007FFF8C561454 (ntdll) : RtlRaiseException
20) 00007FFF8C5B0BFE (ntdll) : KiUserExceptionDispatcher
19) D:\Git\o3de\Gems\ScriptCanvas\Code\Include\ScriptCanvas\Execution\ExecutionObjectCloning.cpp (27) : ScriptCanvas::Execution::CloneSource::Clone
18) D:\Git\o3de\Gems\ScriptCanvas\Code\Include\ScriptCanvas\Execution\Interpreted\ExecutionInterpretedCloningAPI.cpp (38) : ScriptCanvas::Execution::CloneSourceObject
17) 00007FFEF0F03767 (ScriptCanvas.Editor) : luaD_precall
16) 00007FFEF0F0AD26 (ScriptCanvas.Editor) : luaV_execute
15) 00007FFEF0F0309B (ScriptCanvas.Editor) : luaD_callnoyield
14) 00007FFEF0F03A75 (ScriptCanvas.Editor) : luaD_rawrunprotected
13) 00007FFEF0F03310 (ScriptCanvas.Editor) : luaD_pcall
12) 00007FFEF0EFCE11 (ScriptCanvas.Editor) : lua_pcallk
11) D:\Git\o3de\Gems\ScriptCanvas\Code\Include\ScriptCanvas\Execution\Interpreted\ExecutionInterpretedAPI.cpp (565) : ScriptCanvas::Execution::InterpretedSafeCall
10) D:\Git\o3de\Gems\ScriptCanvas\Code\Include\ScriptCanvas\Execution\Interpreted\ExecutionStateInterpretedPure.cpp (62) : ScriptCanvas::ExecutionStateInterpretedPureOnGraphStart::Execute
 9) D:\Git\o3de\Gems\ScriptCanvas\Code\Include\ScriptCanvas\Execution\RuntimeComponent.cpp (70) : ScriptCanvas::RuntimeComponent::Execute
 8) D:\Git\o3de\Code\Framework\AzCore\AzCore\EBus\Internal\BusContainer.h (146) : AZ::Internal::EBusContainer<AZ::EntityEvents,AZ::EntityEvents,1,1>::Dispatcher<AZ::EBus<AZ::EntityEvents,AZ::EntityEvents> >::Event<void (__cdecl AZ::EntityEvents::*)(AZ::Entit
 7) D:\Git\o3de\Code\Framework\AzCore\AzCore\Component\Entity.cpp (213) : AZ::Entity::Activate
 6) D:\Git\o3de\Code\Framework\AzFramework\AzFramework\Entity\GameEntityContextComponent.cpp (230) : AzFramework::GameEntityContextComponent::OnContextEntitiesAdded
 5) D:\Git\o3de\Code\Framework\AzFramework\AzFramework\Entity\EntityContext.cpp (129) : `AzFramework::EntityContext::InitContext'::`2'::<lambda_1>::operator()
 4) D:\Git\o3de\Code\Framework\AzFramework\AzFramework\Entity\SliceEntityOwnershipService.cpp (118) : AzFramework::SliceEntityOwnershipService::AddEntity
 3) D:\Git\o3de\Code\Framework\AzCore\AzCore\EBus\Internal\BusContainer.h (1541) : AZ::Internal::EBusContainer<AzFramework::GameEntityContextRequests,AzFramework::GameEntityContextRequests,0,0>::Dispatcher<AZ::EBus<AzFramework::GameEntityContextRequests,AzFr
 2) D:\Git\o3de\Code\Framework\AzFramework\AzFramework\Spawnable\SpawnableEntitiesManager.cpp (373) : AzFramework::SpawnableEntitiesManager::ProcessRequest
 1) D:\Git\o3de\Code\Framework\AzFramework\AzFramework\Spawnable\SpawnableEntitiesManager.cpp (239) : AzFramework::SpawnableEntitiesManager::ProcessQueue
```

**Attachments**
[Editor.log](https://github.com/o3de/o3de/files/7602132/Editor.log)
[error.log](https://github.com/o3de/o3de/files/7602133/error.log)
[ErrorNoCrash.txt](https://github.com/o3de/o3de/files/7602162/ErrorNoCrash.txt)
[error.zip](https://github.com/o3de/o3de/files/7602189/error.zip)

**Comment by Hsuehnj** [comment]
How to create a new Function in Script Canvas Editor, I haven't find the  'New Function' menu or button.
https://o3deorg.netlify.app/docs/user-guide/scripting/script-canvas/editor-reference/functions/

**Comment by LB-TomaszSerwanski** [comment]
Hi @Hsuehnj, to create a Function you have to add Input Execution Nodelings by either pressing the icon on the Script Canvas Editor toolbar (to the left of Graph Validation icon) or by selecting an input pin of a node which you wish to have in your Function and selecting _Expose_. Then simply save the graph as a regular Script Canvas script and its node will appear in the Node Palette. I have also reported an [issue  (#1244)](https://github.com/o3de/o3de.org/issues/1244) for outdated docs.

**Comment by carlitosan** [comment]
I get this one. The recent change to move InitialializeInterpretedStatics off of asset loading threads, and automatic triggering from loading the runtime assets with the handler means that dependent scripts (like functions) will not have the function called on them. I am fixing this now.

**Comment by carlitosan** [comment]
https://github.com/o3de/o3de/pull/5958

**Comment by Hsuehnj** [comment]
@LB-TomaszSerwanski thanks a lot!

**Comment by carlitosan** [comment]
merged

**Comment by LB-TomaszSerwanski** [comment]
Fix verified in Stabilization_2110 (4d36fb9).

**Comment by LB-KacperKapusta** [comment]
Fix verified using the Linux o3de_installer_2111_1.
